ballerina/observe_utils.bal

```diff
@@ -35,13 +35,13 @@ type TraceInformation record {|
 |};
 
 isolated function addObservabilityMetricsTags(string key, string value) {
-    if metricsEnabled {
+    if observabilityEnabled && metricsEnabled {
         checkpanic observe:addTagToMetrics(key, value);
     }
 }
 
 isolated function addTracingInfomation(TraceObserverContext|TraceInformation traceInformation) {
-    if !tracingEnabled {
+    if !observabilityEnabled || !tracingEnabled {
```

Comment:
No need to check `observabilityEnabled` here as it is checked first when initializing the `tracingEnabled` variable.

```suggestion
    if !tracingEnabled {
```